app/src/main/java/org/mozilla/fenix/compose/Divider.kt

```diff
@@ -0,0 +1,154 @@
+
+package org.mozilla.fenix.compose
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Generic Divider
+ *
+ * @param modifier [Modifier] Used to shape and position the underlying [androidx.compose.material.Divider]
+ * @param color  [Color] of the divider line
+ */
+@Composable
+fun Divider(
+    modifier: Modifier = Modifier,
+    color: Color = FirefoxTheme.colors.borderPrimary,
+) {
+    androidx.compose.material.Divider(
+        modifier = modifier,
+        color = color,
+        thickness = 0.5.dp,
+    )
+}
+
+/**
+ *
+ */
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
+private fun HorizontalDividerPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer1)) {
+            Column {
+                Text(text = "Above the line")
+
+                Divider(modifier = Modifier.padding(vertical = 10.dp, horizontal = 24.dp))
+
+                Text(text = "Below the line")
+            }
+        }
+    }
+}
+
+/**
+ * To use the vertical divider, add Modifier.fillMaxHeight().width(0.5.dp) and
+ * add any additional modifiers after.
+ */
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
+private fun VerticalDividerPreview() {
+    FirefoxTheme {
+        Box(
+            Modifier
+                .background(FirefoxTheme.colors.layer1)
+                .height(75.dp),
+        ) {
+            Row {
+                Text(
+                    text = "Before the line",
+                    modifier = Modifier.padding(end = 10.dp),
+                )
+
+                Divider(
+                    modifier = Modifier
+                        .fillMaxHeight()
+                        .width(0.5.dp)
+                        .padding(vertical = 10.dp),
+                )
+
+                Text(
+                    text = "After the line",
+                    modifier = Modifier.padding(start = 10.dp),
+                )
+            }
+        }
+    }
+}
+
+/**
+ * Example of divider usage in a list menu
+ */
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
+private fun ListDividerPreview() {
```

Comment:
I don't have a strong opinion on this one. It might simply be better to just extend the `HorizontalDividerPreview` with some more `Text` that are in this @Preview instead of having a third @Preview for a divider, which feels a bit unnecessary to me since the divider is fairly straight forward component.

Reply:
simplified the @Preview examples